### PR TITLE
[WIP] Improve performance of fused SDPA/GQA

### PR DESCRIPTION
**Pull request by asimonov1**
This is a draft PR and draft description.
Related to MFDNN-14723.

# Description

This PR adds a fused, FlashAttention-style CPU SDPA/GQA kernel to the DNNL graph backend and the x64 CPU IR support it needs, improving performance of scaled-dot-product-attention (with grouped-query attention) partitions on x64.

## Motivation

The existing CPU SDPA paths leave performance on the table:

- **`sdp_decomp`** decomposes attention into standalone matmul + softmax + reorder primitives, **materializing the full `seq_q × seq_kv` score matrix** and running a whole-row softmax. It also parallelizes only over `batch × num_head_q` behind a `batch·heads > 2·nthr` (OMP-only) gate, so on high-core-count hosts or small batch/decode it is rejected and drops to the generic fallback.
- **`large_partition`** is the generic subgraph fallback: it runs each attention op as a separate full-tensor primitive with a barrier between ops (multi-GB score tensor at prefill).

Both scale poorly at long sequences because of the O(seq²) score materialization and the non-fused softmax.

## What this PR does

Adds `sdp_fused_brgemm_kernel_t`, a **FlashAttention** kernel that tiles over the KV sequence (`kv_block_width = 512`), keeps the two matmuls on BRGEMM, and replaces the intermediate softmax/normalization with a **JIT-compiled online-softmax epilogue**. Per KV tile it runs: `mm1 (QK^T)` → online-softmax epilogue (scale + select-mask + streaming softmax) → `mm2 (P·V)` → accumulator renormalization (`acc = old_coef·acc + pv`). The full score matrix is **never materialized**, scratch is tile-sized and independent of KV length, and there are **no per-head reorders** (BRGEMM strides over the native Q/K/V layout).

Key points:
- The softmax/renorm epilogue is built with the x64 CPU IR framework (`src/cpu/x64/ir`). This PR adds the IR ops it needs: `vsub/vmax/vdiv/vbcast/vhreduce_max`, `veltwise` (via the existing `jit_uni_eltwise_injector` — so `exp` reuses oneDNN's proven Xbyak transcendental), `vblend`, `vcmp_ne_zero`, and `vload_u8` (for the boolean select mask).
- Only `sdp_fused_softmax_ir.hpp` knows the SDPA epilogue math; the IR framework stays generic.
- Parallelizes over `batch × head`; a scalar epilogue fallback is used when AVX2 is unavailable.
- Opt-in via `ONEDNN_GRAPH_SDPA_IMPL=fused_brgemm` (default remains `decomp`). Unsupported cases (quantized int8, additive attention mask, soft-capping, non-strideable layouts) return `unimplemented` and fall back correctly.

## Performance data

**Test input** — both cases are the same f32 GQA attention graph (`MatMul(QK^T) → Multiply(scale) → Select → SoftMax → MatMul(PV)`), strided layout, differing only in query length (from MFDNN-14723):

- batch = 8, num_head_kv = 4, GQA group = 2 (num_head_q = 8), head_size = 256, seq_kv = 2049
- **Decode** = `graph-100254.json`: Q `[8, 4, 2, 1, 256]`, K `[8, 4, 1, 256, 2049]` (seq_q = 1)
- **Prefill** = `graph-100215.json`: Q `[8, 4, 2, 2049, 256]`, same K (seq_q = 2049)

Reproduce with:
```
# prefill
ONEDNN_GRAPH_SDPA_IMPL={decomp|fused_brgemm} benchdnn --graph --mode=P [--cold-cache=all] --case=graph-100215.json
# decode
ONEDNN_GRAPH_SDPA_IMPL={decomp|fused_brgemm} benchdnn --graph --mode=P [--cold-cache=all] --case=graph-100254.json
```
on GNR CPU.

Release build, 16 threads pinned to NUMA node 0, `benchdnn --graph --mode=P`.

**Warm cache**

| Case | decomp | fused (this PR) | speedup |
|---|---|---|---|
| Decode  | 2.604 ms | 1.043 ms | 2.5× |
| Prefill | 1332.29 ms | 124.06 ms | 10.7× |

**Cold cache** (`--cold-cache=all`)

| Case | decomp | fused (this PR) | speedup |
|---|---|---|---|
| Decode  | 2.615 ms | 1.054 ms | 2.48× |
| Prefill | 1354.91 ms | 137.16 ms | 9.9× |

The gain is largest at prefill, where `decomp`'s score-matrix materialization and full-row softmax dominate; at decode it comes from avoiding per-head reorders/primitive-dispatch overhead and the thread-count gate.

## Scope / follow-ups (not in this PR)

f32 only (no quantized int8), boolean select mask only (no additive attention mask), no soft-capping, AVX2 IR backend only. Planned next: AVX-512 IR backend, `kv_block_width` tuning, GQA group-into-M folding, seq_q-block parallelization, and a fused-specific eligibility check replacing the reused decomp gate.

### CODE REVIEW

**Review by dzarukin — CHANGES_REQUESTED** [review]
Referring to the status report:
* AVX2 is out of interest, it should support AVX512+.
* bf16 has higher priority than f32 with AMX enabled.
* GPU algorithm is unapplicable here due to online softmax algorithm is significantly slower than existing solution for CPU.
* A single data point is basically unrepresentable. Use the inputs from the graph folder to ensure they don't regress.
* Changes must not be done on graph side, a proper wiring identical to GPU one should be coded.

Please use human resources for proper designing as this feature intends to be important for TF and PT teams.
